src/train.py

```diff
@@ -62,24 +61,63 @@ def train(args):
         TODO: Include functionality for saving, loading from save
     '''
     set_seed(args.seed)
-    dataset = get_dataset()
+    print("#### Getting Dataset ####")
+    supervised_train_dataset = get_dataset('labeled_train.csv')
+    unsupervised_train_dataset = get_dataset('unlabeled_train.csv')
+    val_dataset = get_dataset('val.csv')
+    full_train_dataset = get_dataset('full_train.csv')
     # init models and optimizers
     model = None
     optimizer = None
 
     for epoch in range(args.epochs):
-        dataloader = DataLoader(dataset, batch_size=args.batch_size, shuffle=True, collate_fn=collate_fn_transformer, drop_last=True, num_workers=16)
+        # Training model
+        supervised_dataloader = DataLoader(supervised_train_dataset, batch_size=args.batch_size, shuffle=True, collate_fn=collate_fn_transformer, drop_last=True, num_workers=16)
+        supervised_iter = iter(supervised_dataloader)
+
+        unsupervised_dataloader = DataLoader(unsupervised_train_dataset, batch_size=args.batch_size, shuffle=True, collate_fn=collate_fn_transformer, drop_last=True, num_workers=16)
+
+        bar = tqdm(unsupervised_dataloader)
+        bar.set_description("Epoch {} Training Model".format(epoch))
+
+        # step counter for a single epoch
+        # used to determine which training task to do
+        epoch_step = 0
+
+        # We are considering one pass through the unsupervised dataset as
+        # one epoch
+        for unsupervised_batch in bar:
+            epoch_step += 1
+            if epoch_step % 3 == 0:
+                # Janky way to check to do a supervised step
+                try:
+                    supervised_batch = supervised_iter.next()
+                except StopIteration:
+                    supervised_iter = iter(DataLoader(supervised_train_dataset, batch_size=args.batch_size, shuffle=True, collate_fn=collate_fn_transformer, drop_last=True, num_workers=16))
+                    supervised_batch = supervised_iter.next()
+                # TODO: Do a supervised step with supervised_batch
 
-        pbar = tqdm(dataloader)
-        for i, data in enumerate(pbar):
-            pbar.set_description("Processing at epoch %d"%epoch)
+                # This enforces that we still use the unsupervised_batch in this
+                # iteration
+                epoch_step += 1
+            if epoch_step % 3 == 1:
+                # TODO: Do a denoising step with this unsupervised_batch
+                pass
+            if epoch_step % 3 == 2:
+                # TODO: Do a unsupervised cross modal step
+                # with this unsupervised_batch
+                pass
 
-            character, mel, mel_input, pos_text, pos_mel, _ = data
+        if args.train_discriminator:
```

Comment:
Should we consider training the discriminator on a subset of the full_train_dataset?  Otherwise it will take quite a while.. maybe use `dataset = torch.utils.data.Subset(dataset, random_indices) ` to get a different random subset of say 1000 examples on each epoch

Reply:
You don't have to create a separate Subset for this, you can simply use
```python3
discriminator_train_examples = 1000
discriminator_dataloader = DataLoader(full_train_dataset, batch_size=args.batch_size,
                                      sampler=np.random.choice(len(full_train_dataset), discriminator_train_examples),
                                      collate_fn=collate_fn_transformer, drop_last=True, num_workers=16)
```